### ISSUE TITLE
Overriding offsetGet in class that implements ArrayAccess throws internal error when parsing to pure+ mode

### ISSUE THREAD

**Issue by robertleeplummerjr** [DEVSENSE/Phalanger#51]
the implementation is very simple:

```php
public class Test implements ArrayAccess {
	public function &offsetGet( $offset )
	{
		return $this->__get( $offset );
	}
}
```

The actual code is here: https://github.com/robertleeplummerjr/redbean/blob/master/RedBeanPHP/OODBBean.php
The command that triggers it: https://github.com/robertleeplummerjr/redbean.net/blob/master/build.bat

The error:
```
error PHP2007: Internal error! 
Please, report this bug via http://phalanger.codeplex.com/workitem/list/basic.
Additional information: 
Method 'offsetGet' in type 'RedBeanPHP.OODBBean' from assembly 'RedBean, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
   at System.Reflection.Emit.TypeBuilder.TermCreateClass(RuntimeModule module, Int32 tk, ObjectHandleOnStack type)
   at System.Reflection.Emit.TypeBuilder.CreateTypeNoLock()
   at System.Reflection.Emit.TypeBuilder.CreateType()
   at PHP.Core.Reflection.PhpType.Bake()
   at PHP.Core.Reflection.PureCompilationUnit.Bake()
   at PHP.Core.Reflection.PureCompilationUnit.Compile(IEnumerable`1 sourceFiles, PureAssemblyBuilder assemblyBuilder, CompilationContext context, Encoding encoding)
   at PHP.Core.Emit.PureAssemblyBuilder.Build(IEnumerable`1 sourceFiles, CompilationContext context)
   at PHP.Core.ApplicationCompiler.Compile(ApplicationContext applicationContext, CompilerConfiguration config, ErrorSink errorSink, CompilationParameters ps)
```

**Comment by robertleeplummerjr** [comment]
I might add, aside from this little issue, the whole thing works stunningly :)

**Comment by robertleeplummerjr** [comment]
After a little search, it looks like this is intended: https://github.com/DEVSENSE/Phalanger/blob/master/Source/Core/SPL/ArrayAccess.cs#L897

Any way actually have this implemented?

**Comment by robertleeplummerjr** [comment]
Any comment on the offsetGet?

**Comment by jakubmisek** [comment]
The test case works for me. Only warning is
test.php(5,22): warning PHP0180: Declaration of Test::offsetGet() should be compatible with that of ArrayAccess::offsetGet()

**Comment by jakubmisek** [comment]
should be fixed in 630fdbd668aa4d92930b66c0f08c41243af0af54

**Comment by robertleeplummerjr** [comment]
Nice!  Thanks Jakub.

On Wed, Dec 30, 2015 at 11:51 AM, Jakub Míšek <notifications@github.com>
wrote:

> Closed #51 <https://github.com/DEVSENSE/Phalanger/issues/51>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DEVSENSE/Phalanger/issues/51#event-502909131>.
>



-- 
Robert Plummer
